lens.cabal

```diff
@@ -412,40 +401,17 @@ test-suite hunit
       test-framework       >= 0.6,
       test-framework-hunit >= 0.2
 
--- Verify the results of the examples
+-- We need this dummy test-suite to add simple-reflect to the install plan
 test-suite doctests
   type:              exitcode-stdio-1.0
   main-is:           doctests.hs
-  ghc-options:       -Wall -threaded
   hs-source-dirs:    tests
   default-language:  Haskell2010
-  x-doctest-options: --fast
 
-  if flag(trustworthy) && impl(ghc>=7.2)
-    other-extensions: Trustworthy
-    cpp-options: -DTRUSTWORTHY=1
-
-  if !flag(test-doctests)
-    buildable: False
-  else
-    build-depends:
-      base,
-      bytestring,
-      containers,
-      directory      >= 1.0,
-      deepseq,
-      doctest        >= 0.11.4 && < 0.12 || >= 0.13 && < 0.18,
-      filepath,
-      generic-deriving,
-      lens,
-      mtl,
-      nats,
-      parallel,
-      semigroups     >= 0.9,
-      simple-reflect >= 0.3.1,
-      text,
-      unordered-containers,
-      vector
+  build-depends:
+    base,
+    deepseq,
+    simple-reflect >= 0.3.1
```

Comment:
Oof, the fact that you have to list dependencies on `deepseq` and `simple-reflect` _twice_ in the same `.cabal` file is rather painful. Is there a way that we can make it so that they only have to be listed once?

Reply:
deepseq is not strictly needed (because it's transitive dependency of about everything). But I think that repeating here is better, then having some non-obvious rule of thumbs.

Note: **if** in the bright future, `cabal-install` learns about `cabal-docspec`, it could learn `x-docspec-extra-packages` (or similar), which then would be `x-docspec-extra-packages`.

i have to think about `x-docspec-extra-packages`. Maybe I'll add it tomorrow if I still think it's not a bad idea.